### PR TITLE
Skip degenerate qsort: num < 2, size == 0, base == 0

### PR DESCRIPTION
**Pull request by jaykrell**
Fixes https://github.com/mono/mono/issues/15994

Some is redundant for some qsort implementations.

### CODE REVIEW

File: mono/utils/mono-qsort.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+#include <stdlib.h>
+
+static inline
+void
+mono_qsort (void* base, size_t num, size_t size, int (*compare)(const void*, const void*))
+{
+	g_assert (compare);
+	g_assert (size);
+	if (num < 2 || !size || !base)
```
Comment (lambdageek) [inline mono/utils/mono-qsort.h]: `!size` is redundant - you assert that size is non-zero on the previous line.

Comment (jaykrell) [reply mono/utils/mono-qsort.h]: Code as if assert might be disabled?

**Comment by jaykrell** [comment]
Some changes:
 moved to glib.h; still called mono_qsort
 restored some of the `if` that guarded some qsort, redundant but maybe worthwhile

**Comment by jaykrell** [comment]
@monojenkins build failed

**Pull request merged**